rust/private/utils.bzl

```diff
@@ -12,43 +12,59 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""
-Utility functions not specific to the rust toolchain.
-"""
+"""Utility functions not specific to the rust toolchain."""
 
 def find_toolchain(ctx):
-    """Finds the first rust toolchain that is configured."""
-    return ctx.toolchains["@io_bazel_rules_rust//rust:toolchain"]
-
-def relative_path(src_path, dest_path):
-    """Returns the relative path from src_path to dest_path."""
-    src_parts = _path_parts(src_path)
-    dest_parts = _path_parts(dest_path)
-    n = 0
-    done = False
-    for src_part, dest_part in zip(src_parts, dest_parts):
-        if src_part != dest_part:
-            break
-        n += 1
+    """Finds the first rust toolchain that is configured.
 
-    relative_path = ""
-    for i in range(n, len(src_parts)):
-        relative_path += "../"
-    relative_path += "/".join(dest_parts[n:])
+    Args:
+        ctx (ctx): The ctx object for the current target.
 
-    return relative_path
+    Returns:
+        ToolchainContext: A Rust toolchain context.
+    """
+    return ctx.toolchains["@io_bazel_rules_rust//rust:toolchain"]
 
 def _path_parts(path):
     """Takes a path and returns a list of its parts with all "." elements removed.
 
-    The main use case of this function is if one of the inputs to _relative()
+    The main use case of this function is if one of the inputs to relativize()
     is a relative path, such as "./foo".
 
     Args:
-      path_parts: A list containing parts of a path.
+      path: A list containing parts of a path.
 
     Returns:
       Returns a list containing the path parts with all "." elements removed.
     """
     path_parts = path.split("/")
     return [part for part in path_parts if part != "."]
+
+def relativize(path, start):
```

Comment:
Why move the position of the method?

Reply:
it reduces the amount of code in `rustc.bzl` which I find valuable since there is a lot going on there.

Reply:
In this change the function simply change position inside utils.bzl. Its not changing file. Side-note: a general good methodology when relocating is not changing the code at all in the same change just moving it so we can easily see when we change something in one function.

Reply:
Ah, sorry, I misread the diff just now. I fixed the positioning and that's a good thing to call out. I'll be better about that in future changes 😄

Reply:
The positioning has been updated. Hopefully this is more clear